recipes/jenkins.rb

```diff
@@ -31,6 +31,53 @@
 apt_update
 
 package 'openjdk-8-jdk-headless'
+# Jenkins downgrade protection
+#
+# The Jenkins package has transitioned to using systemd units instead of
+# sysvinit style init scripts. In order to maintain this cookbook the last
+# version of Jenkins which is packaged with a sysvinit script is pinned.
+# However Jenkins cannot be safely downgraded and some configurations of
+# Jenkins will continue to function once initially configured even after the
+# systemd switch.
+# To protect users from unintentional downgrades we're going to do something really ugly here.
+package 'jenkins' do
+  action :lock
+  only_if { -> do
```

Comment:
The `{` opens the Ruby block that chef will eval at convergence time and then within it I immediately start a lambda literal with `-> do` (affectionately known as the "stabby lambda" back in my day) creating a block within a block.

I need to do this in order to use early returns in the code below since using `return` or `break` in a block literal will result in a LocalJumpError. I'll `call` the lambda as soon as I close it passing the result to the outer block.